### ISSUE TITLE
Disappearing tool heads and addons

### ISSUE THREAD

**Issue by Novtalath** [BluSunrize/ImmersiveEngineering#5055]
MC 1.16.4 
ImmersiveEngineering-1.16.5-5.0.5-140
Modpack: Techopolis A4.1

Buzzsaw and Drill so far.

Added head to drill and some additional modifiers (Tank, lubrication etc)
Then went to add stuff to Buzzsaw and when picked it back from table I have noticed that drill has lost all items from inside (apart from shader what's weird). 
Went to add drill head and my other addons to drill again - after picking it up - Buzzsaw lost all from inside.

**Comment by BluSunrize** [comment]
Were those tools spawned in creative?

**Comment by TheInfamousAlk** [comment]
I have the same thing. However, I also experience this with the engineer's toolbox. It happens when I craft things in survival or spawn them in creative. Whenever I switch between survival and creative, all addons in my tools disappear (except shaders) and everything in my engineer's toolbox disappears.

**Comment by TheInfamousAlk** [comment]
It also happens when I log off and back on to a server.

**Comment by TheInfamousAlk** [comment]
https://github.com/MinecraftForge/MinecraftForge/issues/4580

It looks like the same issue as found here from 1.12.

**Comment by TheInfamousAlk** [comment]
It appears the NBT data is still there temporarily. When I hold alt I can see the NBT data of the tools, and it shows the upgrades in there. They however, are not there and do not work. When I put the tool (drill or buzzsaw) into the engineer's workbench while in this state, the upgrades do not show up. I can still see the upgrades in the NBT data while it is in the workbench. But when I click one of the empty upgrade slots, all of the NBT data is wiped. I do not know if this is helpful or not.

In other words, the tool is wiped, and the NBT data is not, and the NBT data does not match the current "empty" state of the tool, until I click a slot in the workbench, at which point the NBT data is changed to "match" the current empty state of the tool.

**Comment by TheInfamousAlk** [comment]
I am using the modpack all the mods 6 version 1.8.9, and I have submitted a bug to their github as well. But it seems like it is happening elsewhere, not just in that modpack so I'm commenting my experience here. 
 
 The first instance was an engineer's toolbox wiping all items stored inside. I crafted the toolbox, used it for a day (real time), logged off, and the next day I logged on and its contents were empty. I had several items from several mods inside. Not just IE stuff.

The 2nd instance was when I crafted a mining drill and buzz saw (again, since they were wiped) and upgraded them with blades, drill head, and several upgrades and shaders. It worked, and it stored the upgrades just fine. Immediately after, I was just going around killing mobs with my new railgun, when I open my inventory and find that the drill and buzzsaw now have no blade, drill head, or any upgrades. The shaders were the only thing that was not wiped. I did not log off, change game mode from survival, or do anything strange like teleport. I simply crafted them, installed the upgrades, and started walking around killing mobs nearby. (in the log I pasted, the disappearance happened maybe 10-15 minutes before the end of the log, which is when I copied it)

This has not happened to any storage items or tools with internal storage from any other mod while playing the modpack. Just IE.

**Comment by TheInfamousAlk** [comment]
Update: my capacitor backpack keeps deleting all power from it as well. Strangely, my heavy plated shield, with its 2 upgrades, has never been wiped. It has been in my offhand the entire time.

**Comment by Westermin** [comment]
Experiencing the same issue with the drill mods disappearing on version 1.18.1-7.0.0-142

**Comment by BluSunrize** [comment]
Where you using creative mode? Because we've elaborated multiple times now that this causes issues.

**Comment by TheInfamousAlk** [comment]
I was not using creative at all. This happens strictly in survival.

**Comment by Westermin** [comment]
Can confirm, played 100% survival, and it still happens

**Comment by ArchieBeepBoop** [comment]
Can confirm this is happening too in survival. Simply opening up my inventory after adding a fresh drill head causes it to disappear. [1.18.1]

Also, if doing so in creative it will always happen upon adding a new drill head to the drill, setting it on your hotbar and then opening up your inventory.

**Comment by MisterGunpowder** [comment]
This is happening in 1.16.5 on a server set for survival. I installed a buzz saw blade, then remove it and it disappears. If I put it back in, the blade returns, so it's remembering that it's _there_ but isn't able to detect it when it's trying to do anything. I haven't yet tried it in singleplayer to see if it's some server issue, but I'm not sure if that matters. [Here is a link to the installed mods.](https://pastebin.com/QvsXLXcL) Something I did installed that targeted IE is that I installed ProjectE Integrations. Is it possible that having EMC added to the sawblade is messing with it?

**Comment by BluSunrize** [comment]
It's a Forge bug related to item capabilties which we can not work around.

**Comment by TheInfamousAlk** [comment]
If it is a bug that cannot be fixed, and cannot be worked around, then remove the broken feature. If it will remain eternally broken, it doesn't need to exist. Or maybe try an approach like what draconic evolution uses with its upgrade modules.
